pages/index.tsx

```diff
@@ -1,7 +1,114 @@
-import React, { Component } from 'react'
+import Contest from '../components/Contest'
+import Layout from '../components/Layout'
+import { getContestList } from '../services/reventService'
+import React, { useState, useEffect } from 'react'
+import SearchInput from '../components/SearchInput'
 
-export default class index extends Component {
-  render() {
-    return <div>Home Page</div>
+export default function Home(props) {
+  const { contestList } = props
+
+  const [statusFilter, setStatusFilter] = useState('All')
+
+  const [open, setOpen] = useState(false)
+
+  const filteredContestList = contestList.filter(
+    (contest) => statusFilter === 'All' || contest.status === statusFilter
+  )
+
+  const handleStatusFilter = (status: string) => {
+    setStatusFilter(status)
+  }
+
+  useEffect(() => {
+    // When hiding the filters, reset them
+    if (open == false) {
+      setStatusFilter('All')
+    }
+  }, [open])
```

Comment:
does it make sense to reset the filters just because the user hid them?

Reply:
will remove